src/prompt/variable.rs

```diff
@@ -131,106 +133,70 @@ fn quoted_str(input: &str) -> IResult<&str, String, ParsingError> {
-                                    std::char::from_u32
-                                ),
-                                map_res(
-                                    map_res(
-                                        preceded(
-                                            char('x'),
-                                            verify(
-                                                take(2usize),
-                                                |s: &str| s.chars().all(|c| c.is_ascii_hexdigit())
-                                            )
-                                        ),
-                                        move |hex| u8::from_str_radix(hex, 16).context("Invalid hex digit")
-                                    ),
-                                    |digit| {
-                                        if digit > 0x7F || digit == 0 {
-                                            return Err(
-                                                format!(
-                                                    "invalid string literal: \
-                                                     invalid escape sequence '\\x{:x}' \
-                                                     (only non-null ascii allowed)", digit)
-                                            );
-                                        }
-
-                                        Ok(digit as char)
-                                    }
-                                ),
-                                value('\n', char('n')),
-                                value('\r', char('r')),
-                                value('\t', char('t')),
-                                value('\u{08}', char('b')),
-                                value('\u{0C}', char('f')),
-                                value('\\', char('\\')),
-                                value('/', char('/')),
-                                value('"', char('"')),
-                                value('\'', char('\'')),
-                            ))
-                        ),
-                        StringFragment::EscapedChar
-                    ),
-                    value(
-                        StringFragment::EscapedWS,
-                        preceded(char('\\'), nom::character::streaming::multispace1)
-                    ),
+        map_res(
+            recognize(
+                tuple((
+                    char(quote),
+                    move |str: &'a str| {
+                        let mut pos = 0;
+
+                        if str.is_empty() {
+                            return Err(Error(ParsingError::Incomplete));
+                        }
+
+                        if str.len() == 1 && str.chars().nth(0).unwrap() != quote {
+                            return Err(Error(ParsingError::Mistake {
+                                kind: None,
+                                description: "Missing end quote on string".to_string()
+                            }))
+                        }
```

Comment:
This logic shouldn't be necessary for correctness, right? Since if either of those conditions hold, the loop below should exit with `pos == 0` and then the `char(quote)` should cause the error.

Reply:
Seems to be the case, I'll remove the redundant code